auto_route/example/lib/mobile/screens/user-data/single_field_page.dart

```diff
@@ -25,9 +24,11 @@ class _SingleFieldPageState extends State<SingleFieldPage> {
   Widget build(BuildContext context) {
     return WillPopScope(
       onWillPop: () async {
-        _scaffoldKey.currentState?.showSnackBar(
-          SnackBar(content: Text(widget.willPopMessage)),
-        );
+        ScaffoldMessenger.of(_scaffoldKey.currentContext!)
+            .showSnackBar(SnackBar(content: Text(widget.willPopMessage)));
+        // _scaffoldKey.currentState?.showSnackBar(
```

Comment:
I would not recommend using the null assertion operator (!). In general, we can avoid it and write a safer code.

By the way, here let's either remove the commented code or restore the old one. IMO, just looking at it, the old one could be fine since it uses the null-aware operator (?.), so it either runs properly or not at all. 🙂

Reply:
Correct, a better option would be to check null value. but I patched this up hurriedly because I was short on time, and it didn't seem to break anything.